internal/services/netapp/netapp_volume_group_resource.go

```diff
@@ -0,0 +1,642 @@
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"snapshot_policy_id": {
+									Type:         pluginsdk.TypeString,
+									Required:     true,
+									ValidateFunc: azure.ValidateResourceID,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func (r NetAppVolumeGroupResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		/*
+			TODO - This section is for `Computed: true` only items, i.e. useful values that are returned by the
+			datasource that can be used as outputs or passed programmatically to other resources or data sources.
+
+			NOTE: Not applicable for this resource type
+		*/
+	}
+}
+
+func (r NetAppVolumeGroupResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 90 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.NetApp.VolumeGroupClient
+			replicationClient := metadata.Client.NetApp.VolumeReplicationClient
+
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			var model NetAppVolumeGroupModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			id := volumegroups.NewVolumeGroupID(subscriptionId, model.ResourceGroupName, model.AccountName, model.Name)
+
+			metadata.Logger.Infof("Import check for %s", id)
+			existing, err := client.VolumeGroupsGet(ctx, id)
+			if err != nil && existing.HttpResponse.StatusCode != http.StatusNotFound {
+				return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+			}
+
+			if existing.Model != nil && existing.Model.Id != nil && *existing.Model.Id != "" {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			applicationType := volumegroups.ApplicationType(model.ApplicationType)
+
+			volumeList, err := expandNetAppVolumeGroupVolumes(model.Volumes, id)
+			if err != nil {
+				return err
+			}
+
+			// Performing some basic validations that are not possible in the schema
+			if errorList := validateNetAppVolumeGroupVolumes(volumeList, applicationType); len(errorList) > 0 {
+				return fmt.Errorf("one or more issues found while performing deeper validations for %s:\n%+v", id, errorList)
+			}
+
+			// Parse volume list to set secondary volumes for CRR
+			for i, volumeCrr := range *volumeList {
+				if volumeCrr.Properties.DataProtection != nil &&
+					volumeCrr.Properties.DataProtection.Replication != nil &&
+					strings.EqualFold(string(*volumeCrr.Properties.DataProtection.Replication.EndpointType), string(volumegroups.EndpointTypeDst)) {
+
+					// Modify volumeType as data protection type on main volumeList
+					// so it gets created correctly as data protection volume
+					(*volumeList)[i].Properties.VolumeType = utils.String("DataProtection")
+				}
+			}
+
+			// TODO: deploymentSpecId is temporary until the backend is updated and deploymentSpecId is not required anymore,
```

Comment:
This is a TODO for when a major update will happen, where another AVG type will be introduced, I left there so I don't forget to remove it later, it will be in 4-6 months timeframe.